save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityProjectWindow.kt

```diff
@@ -21,100 +22,203 @@ import react.useState
+                    range.type.name,
+                    RangeType.values().map { it.toString() },
+                    classes = "input-group form-control custom-select",
+                ) { event ->
+                    val entries = event.target.value
+                    val typeSelect = requireNotNull(RangeType.valueOf(entries))
+                    val newRange = range.copy(type = typeSelect)
+                    setRange(newRange)
+                    setVulnerabilityProject(vulnerabilityProject.copy(ranges = listOf(newRange)))
+                }
+            }
+
+            inputTextFormOptional {
+                form = InputTypes.VERSION
+                textValue = range.events.first().introduced
+                classes = "col-6 pl-2 pr-2 mt-3 text-left"
+                name = "Introduces version"
+                onChangeFun = { event ->
+                    val newRange = range.copy(events = listOf(range.events.first().copy(introduced = event.target.value)))
+                    setRange(newRange)
+                    setVulnerabilityProject(vulnerabilityProject.copy(ranges = listOf(newRange)))
+                }
+            }
+
+            inputTextFormOptional {
+                form = InputTypes.GIT_URL
+                textValue = vulnerabilityProject.`package`?.introducedCommits?.first() ?: ""
+                classes = "col-6 pl-2 pr-2 mt-3 text-left"
+                name = "Introduces Url commits"
+                onChangeFun = { event ->
+                    val newPackage = vulnerabilityProject.`package`?.copy(introducedCommits = listOf(event.target.value))
+                    setVulnerabilityProject(vulnerabilityProject.copy(`package` = newPackage))
+                }
+            }
+
+            inputTextFormOptional {
+                form = InputTypes.VERSION
+                textValue = range.events.first().fixed ?: ""
+                classes = "col-6 pl-2 pr-2 mt-3 text-left"
+                name = "Version with fix"
+                onChangeFun = { event ->
+                    val newRange = range.copy(events = listOf(range.events.first().copy(fixed = event.target.value)))
+                    setRange(newRange)
+                    setVulnerabilityProject(vulnerabilityProject.copy(ranges = listOf(newRange)))
+                }
             }
-        }
 
-        if (vulnerabilityProject.type != VulnerabilityProjectType.COMMIT) {
             inputTextFormOptional {
-                form = InputTypes.PROJECT_VERSION
-                textValue = vulnerabilityProject.versions.joinToString(" ,")
-                validInput = conflictErrorMessage.isNullOrEmpty()
-                classes = "col-12 pl-2 pr-2 mt-3 text-left"
-                name = "Versions".t()
+                form = InputTypes.GIT_URL
+                textValue = vulnerabilityProject.`package`?.fixedCommits?.first() ?: ""
+                classes = "col-6 pl-2 pr-2 mt-3 text-left"
+                name = "Commits url with fix"
```

Comment:
Please proofread.

At least, _URL_ (capital case) and _with **a** fix_.

Reply:
Fixed